src/Collapsable.jsx

```diff
@@ -0,0 +1,99 @@
+var React = require('react');
+var joinClasses = require('./utils/joinClasses');
+var BootstrapMixin = require('./BootstrapMixin');
+var CollapsableMixin = require('./CollapsableMixin');
+var classSet = require('./utils/classSet');
+var domUtils = require('./utils/domUtils');
+var cloneWithProps = require('./utils/cloneWithProps');
+
+var ValidComponentChildren = require('./utils/ValidComponentChildren');
+var createChainedFunction = require('./utils/createChainedFunction');
+
+
+var Collapsable = React.createClass({
+  mixins: [BootstrapMixin, CollapsableMixin],
+
+  propTypes: {
+    onSelect: React.PropTypes.func,
+    expanded: React.PropTypes.bool,
+    eventKey: React.PropTypes.any
+  },
+
+  getCollapsableDOMNode: function () {
+    return this.getDOMNode();
+  },
+
+  getCollapsableDimensionValue: function () {
+    var height = 0;
+    var nodes = this.refs;
+    for (var key in nodes) {
+      if (nodes.hasOwnProperty(key)) {
+        var n = nodes[key].getDOMNode(), h = n.offsetHeight, computedStyles = domUtils.getComputedStyles(n);
+        height += (h + parseInt(computedStyles.marginTop, 10) + parseInt(computedStyles.marginBottom, 10));
+      }
+    }
+    return height;
+  },
+
+  render: function () {
+
+    var classes = this.props.collapsable ? this.getCollapsableClassSet() : {};
+    classes['navbar-collapse'] = this.props.collapsable;
+
+    return (
+      <div eventKey={this.props.eventKey} className={joinClasses(this.props.className, classSet(classes))} >
+        {ValidComponentChildren.map(this.props.children, (this.props.collapsable) ? this.renderCollapsableChildren : this.renderChildren )}
+      </div>
+    );
+  },
+
+  getChildActiveProp: function (child) {
+    if (child.props.active) {
+      return true;
+    }
+    if (this.props.activeKey != null) {
+      if (child.props.eventKey == this.props.activeKey) {
+        return true;
+      }
+    }
+    if (this.props.activeHref != null) {
+      if (child.props.href === this.props.activeHref) {
+        return true;
+      }
+    }
+
+    return child.props.active;
```

Comment:
Not sure on this one. Let's say a dev was using `this.props.active` for another purpose we would always be setting it to false when cloning the child. The implementation above won't override anything. Side note: `getChildActiveProp()` is present in `Nav.jsx`, `SubNav.jsx` and now `CollapsableNav.jsx` shouldn't this be split out into a util?